pkg/apps/util/util.go

```diff
@@ -782,6 +788,33 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+// RolloutExceededTimeoutSeconds returns true if the current deployment exceeded
+// the timeoutSeconds defined for its strategy.
+// Note that this is different than activeDeadlineSeconds which is the timeout
+// set for the deployer pod. In some cases, the deployer pod cannot be created
+// (like quota, etc...). In that case deployer controller use this function to
+// measure if the created deployment (RC) exceeded the timeout.
+func RolloutExceededTimeoutSeconds(config *deployapi.DeploymentConfig, latestRC *v1.ReplicationController) bool {
+	var timeoutSeconds int64
+	if params := config.Spec.Strategy.RollingParams; params != nil {
+		timeoutSeconds = deployapi.DefaultRollingTimeoutSeconds
+		if params.TimeoutSeconds != nil {
+			timeoutSeconds = *params.TimeoutSeconds
+		}
+	}
+	if params := config.Spec.Strategy.RecreateParams; params != nil {
+		timeoutSeconds = deployapi.DefaultRecreateTimeoutSeconds
+		if params.TimeoutSeconds != nil {
+			timeoutSeconds = *params.TimeoutSeconds
+		}
+	}
+	// For "custom" strategy use the default for recreate strategy.
+	if timeoutSeconds == 0 {
```

Comment:
I don't think this will work. If one of the strategies will have TimeoutSeconds=0 this will overwrite it

Reply:
you should probably switch on strategy type